lib/pure/os.nim

```diff
@@ -831,17 +832,21 @@ iterator walkDirRec*(dir: string, filter={pcFile, pcDir}): string {.
   ## ---------------------   ---------------------------------------------
   ## ``pcFile``              yield real files
   ## ``pcLinkToFile``        yield symbolic links to files
-  ## ``pcDir``               follow real directories
-  ## ``pcLinkToDir``         follow symbolic links to directories
+  ## ``pcDir``               yield real directories
+  ## ``pcLinkToDir``         yield symbolic links to directories
   ## ---------------------   ---------------------------------------------
   ##
   var stack = @[dir]
   while stack.len > 0:
     for k,p in walkDir(stack.pop()):
-      if k in filter:
         case k
-        of pcFile, pcLinkToFile: yield p
-        of pcDir, pcLinkToDir: stack.add(p)
+        of pcFile, pcLinkToFile:
+          if k in filter:
+            yield p
+        of pcDir, pcLinkToDir:
+          stack.add(p)
```

Comment:
but what do you do now, if you don't want to follow symbolic links to directories?
i think you should only add pcLinkToDir to the stack if it is in the filter.

Reply:
Maybe to add one more filter?
```nim
iterator walkDirRec*(dir: string, yieldFilter = {pcFile}, followFilter = {pcDir}): string
```